Scripts/modelsystem.py

```diff
@@ -544,13 +544,23 @@ def _add_internal_demand(self, previous_iter_impedance, is_last_iteration):
             random.seed(zone_param.population_draw)
             self.dm.incmod.predict()
             random.seed(None)
-            fname = "agents"
-            header = "\t".join(self.dm.population[0].attr)
-            self.resultdata.print_line(header, fname)
+            fname0 = "agents"
+            fname1 = "tours"
+            # loop to first person that has tours
+            for person0 in self.dm.population:
+                if isinstance(person0.tours, list):
+                    break
+            # print person and tour attr to files
+            self.resultdata.print_line("\t".join(person0.attr), fname0)
+            self.resultdata.print_line(
+                "\t".join(person0.tours[0].attr), fname1)
```

Comment:
We can get the attribute names directly from `Person` and `Tour`:

````
from datatypes.person import Person
from datatypes.tour import Tour


self.resultdata.print_line("\t".join(Person.attr), fname0)
self.resultdata.print_line("\t".join(Tour.attr), fname1)
````

The loop is hence unnecessary (and probably not even working as intended, `person0.tours` is always a `list`, whether it is empty or not, the pythonical way to check if a `list` is non-empty is simply `if person0.tours`).

Reply:
Thanks! I'm glad that there is a better solution as the one I implemented previously was quite complex way of getting headers for output files.